src/api/queries/index.js

```diff
@@ -0,0 +1,180 @@
+/* notyet
+const validFilterOps = {
+  "=": true,
+  "!=": true,
+  ">": true,
+  ">=": true,
+  "<": true,
+  "<=": true,
+  "starts-with": true,
+  "does-not-start-with": true,
+  exists: true,
+  "does-not-exist": true,
+  contains: true,
+  "does-not-contain": true
+};
+*/
+
+export const unaryFilterOps = {
+  exists: true,
+  "does-not-exist": true
+};
+
+export const validCalculateOps = {
+  NONE: true,
+  COUNT: true,
+  SUM: true,
+  AVG: true,
+  COUNT_DISTINCT: true,
+  MAX: true,
+  MIN: true,
+  P001: true,
+  P01: true,
+  P05: true,
+  P10: true,
+  P25: true,
+  P50: true,
+  P75: true,
+  P90: true,
+  P95: true,
+  P99: true,
+  P999: true,
+  HEATMAP: true
+};
+
+export class Calculation {
+  constructor(column, op) {
+    this.column = column;
+    this.op = op;
+    if (this.op === "COUNT") {
+      this.column = "*";
+    }
+  }
+
+  static fromJSON(c) {
+    return new Calculation(c.column, c.op);
+  }
+}
+export function calculation({ column, op }) {
+  return new Calculation(column, op);
```

Comment:
This seems like a lot of `export`ed ways to make a calculation - seems like all of the following are user-accessible ways to make a `P95(response_time)`:

```javascript
import { Calculation, calculation } from '@libhoney/api/queries';

calculation.P95('response_time');
calculation({op: 'P95', column: 'response_time'});
new Calculation('response_time', 'P95');
```

Should we bless just one (my vote: `calculation.P95('response_time')`) and not export the others? (Or am I missing something and that is already the case?)